examples/dwarfdump.rs

```diff
@@ -65,6 +71,57 @@ impl From<io::Error> for Error {
 
 pub type Result<T> = result::Result<T, Error>;
 
+fn parallel_output<II: IntoIterator, F>(max_workers: usize, iter: II, f: F) -> Result<()>
```

Comment:
Any benefit to using `IntoIterator` instead of `Iterator`? Also please move the bound into the where clause.

Reply:
We could use `Iterator` here and move `into_iter` to the caller, but specifying `IntoIterator` here makes this function more reusable with I think no net increase in complexity.